include/bout/mesh.hxx

```diff
@@ -712,7 +717,7 @@ class Mesh {
   
   GridDataSource *source; ///< Source for grid data
   
-  std::map<CELL_LOC, std::shared_ptr<Coordinates> > coords_map; ///< Coordinate systems at different CELL_LOCs
+  std::map<CELL_LOC, std::unique_ptr<Coordinates> > coords_map; ///< Coordinate systems at different CELL_LOCs
```

Comment:
I'm not sure about this change to `unique_ptr` -- is there a reason for that?

Reply:
@ZedThree We were discussing in the thread on #1392 a while ago. I think the conclusion was that `unique_ptr` makes sense because the `Mesh` should always keep ownership of the `Coordinates` objects, but that we should (eventually?) return references rather than pointers. I'm happy to go with whatever you think best on this though; should I change it back?

Reply:
Also, even if `unique_ptr` is the thing to do, it is not a bugfix, so should be in #1392 only, not in this PR. I'll take it out of this one.

Reply:
I think we can even go further and just store values. It was originally stored as a pointer because we have to make the `Mesh` before making a `Coordinates` so it had to be initialised to `nullptr`. But as we're now storing multiple instances in a `map`, we can initialise the map to empty and hand out pointers for now, and references later.